compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/AnnotationDesugar.java

```diff
@@ -610,11 +611,11 @@ private BLangLambdaFunction defineAnnotations(BLangTypeDefinition typeDef, BLang
             fields = ((BLangStructureTypeNode) typeDef.typeNode).fields;
         }
         for (BLangSimpleVariable field : fields) {
-            BLangLambdaFunction paramAnnotLambda = defineAnnotations(field.annAttachments, field.pos, pkgNode, env,
+            BLangLambdaFunction paramAnnotLambda = defineAnnotations(field.annAttachments, pkgNode, env,
                                                                      pkgID, owner, false);
             if (paramAnnotLambda != null) {
                 if (function == null) {
-                    function = defineFunction(typeDef.pos, pkgID, owner);
+                    function = defineFunction(paramAnnotLambda.pos, pkgID, owner);
                     mapLiteral = ASTBuilderUtil.createEmptyRecordLiteral(function.pos, symTable.mapType);
```

Comment:
Is it correct to pass `function.pos` here?

Reply:
Here we need to pass the annotation lambda's position instead of the typedef position.